cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -483,8 +483,33 @@ def create(self, catalog_item_class, *args, **kwargs):
         Returns:
             An instance of catalog_item_class
         """
+        if kwargs:
+            provider = kwargs.get("provider", None)
+            field_entry_point = kwargs.get("field_entry_point", None)
+            if provider and field_entry_point:
+                """kwargs should not contain value of 'provider' and 'field_entry_point'
+                    while instantiating cat_item. Hence released 'provider' and 'field_entry_point'
+                    elements from kwargs.
+                """
+                kwargs.pop("provider")
+                kwargs.pop("field_entry_point")
         cat_item = self.instantiate(catalog_item_class, *args, **kwargs)
-        view = navigate_to(cat_item, 'Add')
+        view = navigate_to(cat_item, "Add")
+        if field_entry_point:
```

Comment:
This is checking field_entry_point, and then making a reference to `provider.category` - where `provider` could be `None`.  It should be checking both, and per my other comments may not even be necessary here when field_entry_point is filled via `fill_dict`.

Why not also make provider a CloudInfraCatalogItem attribute, that's optional to maintain current implementations?

Reply:
@mshriver 
- I have removed this check and updated the way of filling ```entry_point``` while creating catalog items.
- I have added three  checks for [three values](https://github.com/ManageIQ/integration_tests/pull/7903#discussion_r239798559) as I passed these values using ```kwargs```.
- About ```provider``` object: Now we don't need this in create method.